### ISSUE TITLE
Feature - disable nhentai number in the ComicInfo.xml <Title>

### ISSUE THREAD

**Issue by MrHakisak** [9-FS/nhentai_archivist#29]
Love the work you are doing here 9-FS.

I am seeing that the nhentai ID number is put in the title of the ComicInfo.xml <Title> field.
I would like to disable this, it would make the books look more tidy in a library if it didn't have the number (and also allow more room to display long titles). You wouldn't loose sorting either, because upload date is included in ComicInfo.xml.

-Thank you

**Comment by 9-FS** [comment]
Gude MrHakisak, I agree that having the ID in the `<Title>` field is not ideal. Better would be to use a proper field for the ID only like the `<Number>` field for example. But nHentai Archivist's created library is currently optimised to be hosted by [Komga](https://komga.org/docs/guides/scan-analysis-refresh/#import-metadata-for-cbrcbz-containing-a-comicinfoxml-file) (simply because that's what I use and I don't know a better option) and I couldn't figure out how to search by `<Number>` using Komga.

I'm not a big fan of introducing another setting for edge cases, adding more complexity and potentially library inconsistencies. So if I'm going to make a change to the ComicInfo.xml, I'd like to do it only once to keep breaking changes to a minimum and fixing the situation altogether.

So if you have a suggestion on where to put the nhentai ID instead and still be able to search by it using Komga, I'm very open to that change.

I am currently using Komga and have every work categorised as a oneshot, so each work's individual tags are used and no tags grouped in series. As I've said, putting nHentai ID in `<Number>` does not allow for searching. Another possibility, that has been proposed before, would be to add the ID as a tag, but that would literally create hundreds of thousands of individual tags with 1 entry making the filter by tag sidebar completely unusable. And it wouldn't allow for easily searching by tag either.

**Comment by MrHakisak** [comment]
I am new to Komga and cbz files, but I spent a few hours.

I have tried to run a few tests by modifying the ComicInfo.xml, I have tried replacing the number from the title and put it into different fields: Number, Count, Volume, AlternateNumber, AlternateCount, Summary (which shows on komga but can't be searched), Notes, GTIN.

Some things that do work:
Putting the ID at the end of the title.
Adding the ID as an aditional Writer.
or any of these fields:
Penciller, Letterer, Inker, Editor, Cover, Colorist

This is what a oneshot would look like if some of those fields are filled in:
![Screenshot 2024-09-30 203308](https://github.com/user-attachments/assets/bf951b84-b230-4799-8793-8ad59cdfebae)

You would search for them here like this:
![Screenshot 2024-09-30 203415](https://github.com/user-attachments/assets/5bf736c0-6d04-4944-a3e9-7082a321c94d)

Do with this information how you wish (but let me know if you decide against doing any of these changes so I can just start scraping). Also unless there is a software out there that can batch rename every book (let me know if there is), it would mean you would have to re-scrape everything again.

**Comment by 9-FS** [comment]
Okay I didn't know `<Writer>` had its own search field. This is pretty cool.

I am currently inclined to add the ID in `<Writer>` and add a setting to enable having ID additionally in the title just like now, but defaulted to off.

**Comment by shinji257** [comment]
Please don't add the ID into Writer.  I have Writer mapped to artist in LANraragi right now and I already modified the file to account for it.  As it stands I already have a mini patch that changes how nhentai_archivist writes out the Web field because the nHentai metadata plugin over on LANraragi doesn't expect it to have a protocol set at the start.

As it stands the ComicInfo plugin over there isn't able to get the tags from your ComicInfo file.  It's probably a bug.  One that I have not gone through the trouble to fix so I just pull from nHentai in batches.

With all that said I know you are targeting Komga so if you do make a change then I'll adjust depending on the change (if any).

**Comment by 9-FS** [comment]
Hmm you're making a good point though. `<Writer>` is already being used by artists and mixing the field with ID is probably not a good idea.

Komga also offers the fields "pencillers", "inkers", "letterers", "cover", "editors", and "translators". Except for "translators", I'm currently using none of the other fields which could then be used to save the ID. This feels pretty random and unexpected though... I wish they would just allow searching by `<Number>`.

**Comment by MrHakisak** [comment]
[I opened a feature request on komga git](https://github.com/gotson/komga/discussions/1723). They seem kinda active so let's see what they do

**Comment by 9-FS** [comment]
I'm getting crazy bot vibes from the Komga developer to be honest, I think it's time to prepare a plan B in case he's not going to make any changes.

> Komga also offers the fields "pencillers", "inkers", "letterers", "cover", "editors", and "translators". 

We could still do this, but this feels more like a hack than a solution.

**Comment by MrHakisak** [comment]
I was watching, it was strange and unfortunate.
Thank you so much for trying. It's not the end of the world.

I am okay with putting the ID in any of those fields, I agree it feels like a hack. That is okay with me, whatever you are comfortable with. Thank you for trying to be accommodating.

**Comment by 9-FS** [comment]
Yeah he just closed and locked the discussion after I pointed out that his suggestions aren't helpful and he even said so himself; certified bot behaviour. 

I'm thinking about migrating to LANragi altogether as Komga is generally incredibly slow and unreliable. @shinji257 Any tips?

**Comment by shinji257** [comment]
Here are the points I'm hitting.  Currently working with it as it stands myself but doesn't mean that others want to.

* LANraragi has long load times with large libraries.  I dunno.  Probably 25k+ or something.
* ComicInfo plugin needs work to be able to pull in all the metadata from your program.  Currently my version is half baked as it doesn't do tags but I managed to patch some of the other stuff to align.

Once it does load the library it is generally pretty snappy imo.

**Comment by 9-FS** [comment]
Oof no tags are a problem.

**Comment by WorldTeacher** [comment]
adding my two cents here since you mentioned LANraragi. 

I'm currently archiving nhentai with your tool + LANraragi. Some comicinfo.xml files do indeed not get read, can't say if that is because I run the plugin at import or because of some other reason. 
What I noticed is like shinji257 mentioned, it does take a while to load, especially at startup. In my case, I am also unable to tag archives using the built in batch operations, as the application stops the process. While I have not yet opened an issue about that it is somewhat annoying that the tagging is not that reliable

**Comment by shinji257** [comment]
This is the main reason why I use the nhentai plugin to tag the archives.  I made some adjustments in my ComicInfo.  Specifically the following alterations were made partially to accomidate the nhentai plugin.  The other accomidation was the edit I did in this program so that the URL format is what that plugin expects.

ComicInfo is set to give the Genre the category metadata prefix
artist is defined by the Writer ComicInfo tag
group is defined by the Publisher ComicInfo tag
The tag import code is completely disabled in my copy since it is unreliable/broken right now.

All of this prevents duplicate tags like the artist ending up in both the artist and group tag thanks to the default ComicInfo code.

I know that per the ComicInfo definition that Komga has it is pretty flexible on this.

**Comment by WorldTeacher** [comment]
I was thinking about creating a small script that interacts with the db and the api from lanraragi to get untagged files and query the db to get the tags. Will see how that goes, otherwise I'll have to re-tag my library

**Comment by shinji257** [comment]
Proposal for Komga as I was mucking with the ComicInfo.xml import for it.

Move the source id to a tag.  Make it source: {id} or nhentai: {id}.  This is searchable by Komga by doing tag: {id} and it will locate it.

**Comment by shinji257** [comment]
Got a test implentation done.  Probably not the best way as I'm very much a beginner on Rust but I'll still have it on my branch to be looked at.

**Comment by shinji257** [comment]
> Oof no tags are a problem. I'll try it out myself as soon as I have the time.

I'm incorrect on this.  I thought tags were not working but they are working just fine.  The only thing is that categories do not have a category prefix but eh.  At least it is there.  I want the upload timestamp and I can't figure out how to convert the Month/Day/Year back to epoch for LANraragi to handle so I went into your code and did a non-standard Timestamp field and have LANraragi's ComicInfo plugin referencing it in my edits.  It also doesn't set titles so I changed that too... I think.
